AirportItlwm/AirportSTAIOCTL.cpp

```diff
@@ -723,16 +746,20 @@ IOReturn AirportItlwm::
 getDEAUTH(OSObject *object,
                           struct apple80211_deauth_data *da) {
     da->version = APPLE80211_VERSION;
-    da->deauth_reason = APPLE80211_REASON_UNSPECIFIED;
+    ieee80211com *ic = fHalService->get80211Controller();
+    da->deauth_reason = ic->ic_deauth_reason;
+    XYLog("%s, %d\n", __FUNCTION__, da->deauth_reason);
     return kIOReturnSuccess;
 }
 
 IOReturn AirportItlwm::
 getASSOCIATION_STATUS(OSObject *object, struct apple80211_assoc_status_data *hv) {
 //    XYLog("%s\n", __FUNCTION__);
+    ieee80211com *ic = fHalService->get80211Controller();
     memset(hv, 0, sizeof(*hv));
     hv->version = APPLE80211_VERSION;
-    hv->status = APPLE80211_STATUS_SUCCESS;
+    hv->status = ic->ic_assoc_status;
+    XYLog("%s, %d\n", __FUNCTION__, hv->status);
```

Comment:
I think this log should be remove, it will be called for several times when associating.

Reply:
Yes, can be removed or replaced with a DPRINTF. It was left from debugging.